presto-server-rpm/src/main/rpm/preinstall

```diff
@@ -21,7 +23,7 @@ check_if_correct_java_version() {
   JAVA_VERSION=$(java_version "$1")
   JAVA_VENDOR=$(java_vendor "$1")
   JAVA_UPDATE=$(echo $JAVA_VERSION | cut -d'_' -f2)
-  if [[ ("$JAVA_VERSION" > "1.8") && ($JAVA_UPDATE -ge 40) && ("$JAVA_VENDOR" = "Oracle Corporation") ]]; then
+  if  [[ ( -n "$1" ) && ("$JAVA_VERSION" > "1.8") && ($JAVA_UPDATE -ge 40) && ("$JAVA_VENDOR" = "Oracle Corporation") ]] ; then
```

Comment:
Can we please move the test for an empty $1 to the top of the function and add a comment explaining why we return a non-zero result for empty strings?